File: app/src/main/java/com/starsep/sokoban/MainMenuActivity.java
package com.starsep.sokoban;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.starsep.sokoban.database.DatabaseManager;

public class MainMenuActivity extends Activity {
	private Button continueGameButton;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main_menu);
		continueGameButton = (Button) findViewById(R.id.continueGameButton);
	}

	@Override
	protected void onStart() {
		super.onStart();
		continueGameButton.setVisibility(DatabaseManager.instance(this).getCurrentGame(this) == null ? View.GONE : View.VISIBLE);
	}

	public void newGameButtonClicked(View view) {
		Intent intent = new Intent(this, GameActivity.class);
		intent.putExtra("New", true);
		startActivity(intent);
	}

	public void helpButtonClicked(View view) {
		// Intent intent = new Intent(this, HelpActivity.class);
		// startActivity(intent);
	}

	public void settingsButtonClicked(View view) {
		// Dialog settingsDialog = SettingsDialog.build(getApplicationContext());
		// settingsDialog.show();
	}

	public void continueGameButtonClicked(View view) {
		Intent intent = new Intent(this, GameActivity.class);
		intent.putExtra("New", false);
		startActivity(intent);
	}
}

File: app/src/main/java/com/starsep/sokoban/release/activity/MainMenuActivity.kt
package com.starsep.sokoban.release.activity

import android.app.Dialog
import android.content.Intent
import android.os.Bundle
import android.support.annotation.CallSuper
import android.view.View

import com.starsep.sokoban.release.R
import com.starsep.sokoban.release.Sokoban
import com.starsep.sokoban.release.database.Database
import com.starsep.sokoban.release.view.HelpDialogBuilder
import kotlinx.android.synthetic.main.activity_main_menu.*

class MainMenuActivity : SokobanActivity() {
    private var helpDialog: Dialog? = null

    @CallSuper
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setupUI()
    }

    private fun setupUI() {
        setContentView(R.layout.activity_main_menu)
        newGameButton.setOnClickListener {
            val intent = Intent(this, ChooseLevelActivity::class.java)
            startActivity(intent)
        }
        helpButton.setOnClickListener {
            helpDialog = HelpDialogBuilder.build(this)
            helpDialog?.show()
        }
        settingsButton.setOnClickListener {
            startActivity(Intent(this, SettingsActivity::class.java))
        }
        continueGameButton.setOnClickListener {
            val intent = Intent(this, GameActivity::class.java)
            intent.putExtra(Sokoban.NEW, false)
            startActivity(intent)
        }
    }

    override fun onResume() {
        super.onResume()
        Database.getCurrentGame(baseContext)?.let {
            continueGameButton.visibility = View.VISIBLE
        }
    }

    override fun onStop() {
        super.onStop()
        helpDialog?.dismiss()
    }
}
